failmsg.go

```diff
@@ -82,3 +84,11 @@ func (msg FailureMessage) Prefix(s string) FailureMessage {
 	}
 	return FailureMessage(s) + msg
 }
+
+// AsError returns the failure message as an error type, or nil if the message is empty.
+func (msg FailureMessage) AsError() error {
```

Comment:
How about renaming the method to `Err`. The name should be more idiomatic. See e.g.
-  https://pkg.go.dev/context#Context 
- https://pkg.go.dev/bufio#Scanner.Err
- https://pkg.go.dev/database/sql#Rows.Err